tests/countries/test_germany.py

```diff
@@ -9,34 +9,31 @@
 #  Website: https://github.com/dr-prodigy/python-holidays
 #  License: MIT (see LICENSE file)
 
-import unittest
-from datetime import date
-from itertools import product
+from holidays.countries.germany import Germany, DE, DEU
+from tests.common import TestCase
 
-import holidays
 
-
-class TestDE(unittest.TestCase):
-    def setUp(self):
-        self.holidays = holidays.DE()
-        self.prov_hols = {
-            prov: holidays.DE(subdiv=prov) for prov in holidays.DE.subdivisions
+class TestDE(TestCase):
+    @classmethod
+    def setUpClass(cls):
+        years = range(1991, 2050)
+        super().setUpClass(DE, years=years)
+        cls.prov_hols = {
+            prov: DE(subdiv=prov, years=years) for prov in DE.subdivisions
         }
 
+    def test_country_aliases(self):
+        self.assertCountryAliases(Germany, DE, DEU)
+
     def test_no_data_before_1990(self):
-        de_1989 = sum(
-            holidays.DE(years=[1989], subdiv=p)
-            for p in holidays.DE.subdivisions
-        )
+        self.assertNoHolidays(DE(years=1989))
+        de_1989 = sum(DE(years=1989, subdiv=p) for p in DE.subdivisions)
         self.assertEqual(len(de_1989), 0)
 
     def test_all_holidays_present(self):
-        de_2015 = sum(
-            holidays.DE(years=[2015], subdiv=p)
-            for p in holidays.DE.subdivisions
-        )
-        in_2015 = sum((de_2015.get_list(key) for key in de_2015), [])
-        all_de = [
+        de_2015 = sum(DE(years=2015, subdiv=p) for p in DE.subdivisions)
+        y_2015 = set(de_2015.values())
+        all_h = {
```

Comment:
I guess these can be sorted alphabetically.

Reply:
I think the chronological order is more natural here. 🙄